artifact/v3/manifest/manifest.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <artifact/v3/manifest/manifest.hpp>
+
+#include <cctype>
+#include <iterator>
+#include <string>
+#include <sstream>
+#include <regex>
+
+#include <common/error.hpp>
+
+#include <artifact/error.hpp>
+
+namespace mender {
+namespace artifact {
+namespace v3 {
+namespace manifest {
+
+namespace io = mender::common::io;
+namespace error = mender::common::error;
+
+struct ManifestLine {
+	const string shasum;
+	const string entry_name;
+};
+
+using ExpectedManifestLine = expected::expected<ManifestLine, error::Error>;
+
+const size_t expected_shasum_length {64};
+const size_t expected_whitespace {2};
+const string manifest_line_regex_string {
+	"^([[:alnum:]]{" + to_string(expected_shasum_length) + "})[[:space:]]{"
+	+ to_string(expected_whitespace) + "}([/.[:alnum:]]+)$"};
+
+ExpectedManifestLine tokenize(const string &line) {
```

Comment:
Should be `Tokenize` and `static` (if you want to mark it as local).

Reply:
Made it an anonymous namespace